docs/integrations/builtin/credentials/hubspot.md

```diff
@@ -6,70 +6,89 @@ contentType: integration
+
+/// note | API key deprecated
+HubSpot deprecated the API key authentication method. The option still appears in n8n, but you should use the other authentication methods.
 ///
 
+## Related resources
+
+Refer to [HubSpot's API documentation](https://developers.hubspot.com/docs/api/overview){:target=_blank .external-link} for more information about the service. The [HubSpot Trigger](/integrations/builtin/trigger-nodes/n8n-nodes-base.hubspottrigger/) node uses the Webhooks API; refer to [HubSpot's Webhooks API documentation](https://developers.hubspot.com/docs/api/webhooks){:target=_blank .external-link} for more information about that service.
+
+## Using API app token
+
+To configure this credential, you'll need:
+
+- An API **APP Token**: To generate an app token, create a private app in HubSpot. Refer to the [HubSpot Private Apps documentation](https://developers.hubspot.com/docs/api/private-apps){:target=_blank .external-link} for detailed instructions.
+
+## Using Developer API key
+
+To configure this credential, you'll need:
+
+- A **Client ID**: Generated once you create a public app, displayed in the app's **Auth** settings. Refer to the instructions in the [HubSpot Public Apps documentation](https://developers.hubspot.com/docs/api/creating-an-app){:target=_blank .external-link}.
+    - Use the n8n **OAuth Redirect URL** as the **Redirect URL** in your HubSpot app.
+- A **Client Secret**: Generated once you create a public app, displayed in the app's **Auth** settings.
+- A **Developer API Key**: Generated from your Developer Applications dashboard. Refer to [HubSpot Developer API keys](https://legacydocs.hubspot.com/docs/faq/developer-api-keys){:target=_blank .external-link} for more information.
+- An **App ID**: Generated once you create a public app, displayed in the app's **Auth** settings.
+
+Authenticating the credential can fail if you don't use appropriate scopes. See [Recommended scopes for HubSpot Trigger node](#recommended-scopes-for-hubspot-trigger-node) for n8n's suggested scopes for this app.
+
+## Using OAuth2
+
+--8<-- "_snippets/integrations/builtin/credentials/cloud-oauth-button.md"
+
+If you need to configure OAuth2 from scratch, you'll need to create a public app. Refer to the instructions in the [HubSpot Public Apps documentation](https://developers.hubspot.com/docs/api/creating-an-app){:target=_blank .external-link}. If you need more detail on what's happening in the OAuth web flow, refer to the [HubSpot Working with OAuth documentation](https://developers.hubspot.com/docs/api/working-with-oauth){:target=_blank .external-link}.
+
+Authenticating the credential can fail if you don't use appropriate scopes. See [Recommended scopes for HubSpot node](#recommended-scopes-for-hubspot-node) and/or [Recommended scopes for HubSpot Trigger node](#recommended-scopes-for-hubspot-trigger-node) for suggested scopes for this app.
+
+## Recommended scopes for HubSpot node
```

Comment:
I don't think these are just recommended, I think they're required. Can you rephrase at appropriate points to make clear they need to grant these scopes.

Are these just for OAuth, or are they for API App Token as well? If they're just for OAuth, I would make them heading level 3 so they nest within the OAuth section.

Reply:
Good call. Adjustments made:
- All scopes shifted from recommended/suggested to required
- Added required scopes to the App token situation (original instructions didn't include these, but you're right--they'd both need it)
- Removed the required scopes for trigger node references from OAuth2 (and the App token)
- Since the trigger node really only works with the Developer API auth method, I shifted the required scopes for trigger node to a heading level 3 and moved it within the Developer API method. I left the Hubspot node required scopes as heading level 2 since it's now referenced by both App token and OAuth2.